packages/openapi3/test/http-specs/specs.test.ts

```diff
@@ -0,0 +1,164 @@
+import { resolvePath } from "@typespec/compiler";
+import { findTestPackageRoot } from "@typespec/compiler/testing";
+import { deepStrictEqual, ok, strictEqual } from "assert";
+import { exec, execSync } from "child_process";
+import { afterAll, beforeAll, describe, it } from "vitest";
+import { worksFor } from "./../works-for.js";
+import {
+  checkServe,
+  defineSpecTests,
+  markCoverage,
+  validataDataWithSchema,
+} from "./utils/spec-snapshot-testing.js";
+
+const pkgRoot = await findTestPackageRoot(import.meta.url);
+const specsRoot = resolvePath(pkgRoot, "node_modules", "@typespec", "http-specs", "specs");
+const rootOutputDir = resolvePath(pkgRoot, "test", "http-specs", "output");
+
+describe.skip("http-specs convert", () => {
+  defineSpecTests({
+    specDir: specsRoot,
+    outputDir: rootOutputDir,
+    // Excluded cases hits errors/warnings during conversion.
+    exclude: [
+      // warning @typespec/openapi3/invalid-format: Collection format 'tsv' is not supported in OpenAPI3 query parameters. Defaulting to type 'string'.
+      // > 49 |     colors: string[],
+      "parameters/collection-format",
+
+      // warning @typespec/openapi3/xml-unwrapped-invalid-property-type: XML `@unwrapped` can only used on array properties or primitive ones in the OpenAPI 3 emitter, Property 'content' will be ignored.
+      // > 70 |   @unwrapped content: string;
+      "payload/xml",
+
+      // error @typespec/openapi3/unsupported-status-code-range: Status code range '494 to '499' is not supported. OpenAPI 3.0 can only represent range 1XX, 2XX, 3XX, 4XX and 5XX. Example: `@minValue(400) @maxValue(499)` for 4XX.
+      // 59 | model ErrorInRange {
+      "response/status-code-range",
+
+      // error @typespec/openapi3/path-query: OpenAPI does not allow paths containing a query string.
+      // > 435 |       op array(param: string[]): void;
+      "routes",
+    ],
+  });
+});
+
+describe("http-specs cases", () => {
+  worksFor(["3.0.0", "3.1.0"], ({ openApiForFile }) => {
+    describe("AuthApiKeyClient Rest Client", () => {
+      beforeAll(async () => {
+        exec("pnpm spector-serve", { cwd: pkgRoot });
+        await checkServe();
```

Comment:
why do we need to start the server

Reply:
@timotheeguerin, we use the response as ground truth:
```
          strictEqual(response.status, 403);
          strictEqual(response.body.error, "invalid-api-key");
          await validataDataWithSchema(response.body, invalidAuthSchemas);
```